### PR TITLE
Add binary literals to parser

### PR DESCRIPTION
**Pull request by picimaci**
Added binary literals to grammar and code. Did it best to my knowledge however I probably got things wrong at places. Thank you in advance for reviewing :)
Will solve #185 .

### CODE REVIEW

**Comment by benjamin-bader** [comment]
Hi, thanks for the PR, I'm looking forward to reviewing it!  Looks like the unit tests failed in a way that's unrelated to your change, so I restarted them.  It's a busy time right now, so thanks for your patience w/review.

**Comment by benjamin-bader** [comment]
The ANTLR looks perfectly proficient, though I wonder about the definition of the BINARY token.  As defined, it overlaps with STRING and consequently it looks to me like it might either never be matched, or might break some valid string constants.  I'd like to see some unit tests in `LoaderTest` that exercise the whole loader:

```thrift
// these would each be separate test cases
const string STRING_WITH_BINARY_LITERAL = "0xDEADBEEF" // should load

const binary VALID_BINARY = "0xDEADBEEF" // should load

const binary INVALID_BINARY = "blablabla" // should not load
```

I might be wrong here.  If what I suspect is true, the solution will be to drop the changes to the ANTLR grammar, and simply use a STRING token; we'd have to validate its format by hand, in the BinaryValidator.

**Comment by picimaci** [comment]
Or by the next couple of days, sorry for the delay.

**Comment by benjamin-bader** [comment]
Hey, don't feel pressured - there's no rush.  Take your time!

**Comment by picimaci** [comment]
Hey! Any feedback? :)

**Comment by benjamin-bader** [comment]
Sorry, got busy with travel over the past few weeks, and haven't had the time yet to look at the update.  I'll take a look now.

File: thrifty-schema/src/main/kotlin/com/microsoft/thrifty/schema/Constant.kt [hunk 1]
```diff
@@ -239,6 +231,21 @@ class Constant private constructor (
         }
     }
 
+    private object BinaryValidator : BaseValidator() {
+        override fun validate(symbolTable: SymbolTable, expected: ThriftType, valueElement: ConstValueElement) {
+            when (valueElement) {
```
Comment (benjamin-bader) [inline thrifty-schema/src/main/kotlin/com/microsoft/thrifty/schema/Constant.kt]: This `when` needs an `else -> super.validate(symbolTable, expected, valueElement)`.  Without it, we won't properly validate identifiers of binary constants.  We should have a test case for that, too:

```thrift
const binary FOO = "0xDEADBEEF"

struct S {
  1: required binary b = FOO;
}
```

Comment (picimaci) [reply thrifty-schema/src/main/kotlin/com/microsoft/thrifty/schema/Constant.kt]: Totally forgot about this, I'll fix it during the holidays.

**Comment by benjamin-bader** [comment]
Hi there - this PR is nearly ready to land.  Do you still have interest in finishing it?

**Comment by benjamin-bader** [comment]
Hello again.  Any interest in finishing this?  I'll close it out if not.

**Pull request closed without merging**